### ISSUE TITLE
Creating recipe for non-CMake library?

### ISSUE THREAD

**Issue by thoughton** [pfultz2/cget#40]
Hi,

Is there any advice/examples available of best practices for creating a cget recipe for a library that does not already use CMake?

Is there perhaps a way I can "inject" a custom CMakeLists.txt file as part of the recipe? (Or is that thinking about things in the wrong way?)

Thanks!

**Comment by pfultz2** [comment]
> Is there any advice/examples available of best practices for creating a cget recipe for a library that does not already use CMake?

There is a bunch of recipes [here](https://github.com/pfultz2/cget-recipes) which includes recipes for a lot of non-cmake projects, like [xz](https://github.com/pfultz2/cget-recipes/tree/master/recipes/xz) or [wgtdkp/juson](https://github.com/pfultz2/cget-recipes/tree/master/recipes/wgtdkp/juson).

> Is there perhaps a way I can "inject" a custom CMakeLists.txt file as part of the recipe? (Or is that thinking about things in the wrong way?)

Yes, the `-X` or `--cmake` specifies a CMakeLists.txt file to be "injected" into the project.

**Comment by thoughton** [comment]
Thanks for the quick reply, and thank you for the information! I will check out all of those links.

By the way, I have an embarrassing admission: I did read the documentation, but I mistakenly thought it finished at the end of the "Introduction" section! I've only just realised there's a couple more sections to it - which no doubt cover the information you've provided me...

So apologies! I will make sure I study the rest of the documentation now.
